File: client/java/src/avt/EffectObj.java
package avt;

import javax.microedition.lcdui.Graphics;
import main.Canvas;

public final class EffectObj extends Base {
   public short ID;
   public short dx;
   public short dy;
   public int idPlayer;
   public byte style;
   private byte index;

   public EffectObj() {
      this.dx = this.dy = 0;
      super.catagory = 6;
      this.index = 0;
   }

   public final void update() {
      EffectData var1;
      if ((var1 = AvatarData.getEffect(this.ID)) != null) {
         ++this.index;
         if (this.index < var1.arrFrame.length) {
            return;
         }
      }

      this.removee();
   }

   public final void paint(Graphics var1) {
      EffectData var2;
      if ((Canvas.stypeInt <= 0 || Canvas.currentMyScreen != MainMenu.gI()) && (var2 = AvatarData.getEffect(this.ID)) != null) {
         if (this.style == 0) {
            Avatar var3;
            if ((var3 = LoadMap.getAvatar(this.idPlayer)) == null) {
               this.removee();
               return;
            }

            super.x = var3.x + this.dx;
            super.y = var3.y + this.dy;
         }

         var2.paint(var1, super.x, super.y, this.index);
      }

   }

   private void removee() {
      switch (this.style) {
         case 0:
            LoadMap.playerLists.removeElement(this);
            return;
         case 1:
            LoadMap.treeLists.removeElement(this);
            return;
         case 2:
            LoadMap.effBgList.removeElement(this);
            return;
         case 3:
            LoadMap.effCameraList.removeElement(this);
         default:
      }
   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/EffectObj.cs
using System;

public class EffectObj : Base
{
	public short ID;

	public short dx;

	public short dy;

	public int idPlayer;

	public sbyte style;

	public new sbyte index;

	public EffectObj()
	{
		dx = (dy = 0);
		catagory = 6;
		index = 0;
	}

	public override void update()
	{
		try
		{
			EffectData effect = AvatarData.getEffect(ID);
			if (effect != null)
			{
				index++;
				if (index >= effect.arrFrame.Length)
				{
					removee();
				}
			}
			else
			{
				removee();
			}
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}

	public override void paint(MyGraphics g)
	{
		if (Canvas.currentMyScreen == MainMenu.gI())
		{
			return;
		}
		EffectData effect = AvatarData.getEffect(ID);
		if (effect == null)
		{
			return;
		}
		if (style == 0)
		{
			Avatar avatar = LoadMap.getAvatar(idPlayer);
			if (avatar == null)
			{
				removee();
				return;
			}
			x = avatar.x + dx;
			y = avatar.y + dy;
		}
		effect.paint(g, x, y, index);
	}

	private void removee()
	{
		switch (style)
		{
		case 0:
			LoadMap.playerLists.removeElement(this);
			break;
		case 1:
			LoadMap.treeLists.removeElement(this);
			break;
		case 2:
			LoadMap.effBgList.removeElement(this);
			break;
		case 3:
			LoadMap.effCameraList.removeElement(this);
			break;
		}
	}
}
